### DISCUSSION TITLE
How to get larger edge drop zones

### DISCUSSION THREAD

**Discussion by hahughes** [xyflow/xyflow#1180] [Q&A]
The edge drop zones for each node are too pixel perfect small. I want users to be able to drag and drop a source edge anywhere on the target node and have the two connect. Is there an easy way to do this in react-flow? 

I'm using [the "Basic" example](https://github.com/wbkd/react-flow/blob/main/example/src/Basic/index.tsx) to sandbox on.

Bonus points for being able to increase the source edge selection zone!

**Comment by stevegg** [comment]
I don't have a solution but would also like any pointers on this.  As well, has anyone found a way to make selecting edges a little more forgiving?

**Reply by visualsitemaps** [reply]
ditto. +1

**Comment by BenBrewerBowman** [comment]
It took a bit of work and isn't the most straightforward solution, but I was able to solve this:

Styles and number values will need to be adjusted, but here's the idea.

Increase size of handles and add an invisible border for "padded" edge handles:
```
      <Handle
        type="target"
        position={Position.Top}
        style={{
          left: "50%",
          background: "#B1B1B7",
          height: 60,
          width: 60,
          marginTop: -26,
          border: "22px solid rgba(0,0,0,0)",
          backgroundClip: "padding-box",
        }}
      />
      <Handle
        type="source"
        position={Position.Bottom}
        style={{
          left: "50%",
          background: "#B1B1B7",
          height: 60,
          width: 60,
          marginBottom: -26,
          border: "22px solid rgba(0,0,0,0)",
          backgroundClip: "padding-box",
        }}
      />
```
      
The above mostly works, but edge connection lines will only go to the outside of the invisible border radius. We need a custom connection line to draw from the visible portion of the handles:
```
<ReactFlow
    edgeTypes={{ custom: CustomEdgeConnectionLine }}
    ...
/>
```

CustomEdgeConnectionLine
```
import { EdgeProps, getBezierPath, getMarkerEnd } from "react-flow-renderer";

export const ConversationFlowEdge = ({
  sourceX,
  sourceY,
  targetX,
  targetY,
  arrowHeadType,
  markerEndId,
}: EdgeProps) => {
  return (
    <g>
      <path
        fill="none"
        stroke="#B1B1B7"
        strokeWidth={3}
        d={getBezierPath({
          sourceX,
          sourceY: sourceY - 16,
          targetX,
          targetY: targetY + 15,
        })}
        markerEnd={getMarkerEnd(arrowHeadType, markerEndId)}
      />
    </g>
  );
};
```

**Comment by BigBearWayne** [comment]
You can change the size by using css to override the style. 
I increased the diameter to 12px and updated its offset (radius+1).

.react-flow__handle {
  height: 12px !important;
  width: 12px !important;
}

.react-flow__handle-right {
  right: -7px !important;
}

.react-flow__handle-left {
  left: -7px !important;
}

.react-flow__handle-top {
  top: -7px !important;
}

.react-flow__handle-bottom {
  bottom: -7px !important;
}

**Comment by devbro1** [comment]
If there was an API that could tell us that an edge connection is being made, it would allow making handle bigger

```
onConnecting(node: Node, handle: Handle) => {
    if(handle === null)
    {
        setBiggerHandles('none');
    }
    else if(handle.type === 'source')
    {
        setBiggerHandles('target');
    }
    else //if connecting is stopped/failed/finished
    { 
        setBiggerHandles('source');
    }
}
```

then
```
<Handle type="target" className={biggerHandles === 'target'? 'w-2 h-2':''} position={Position.Top} />
<Handle type="target" className={biggerHandles === 'source'? 'w-2 h-2':''} position={Position.Top} />
```

Unless there already exists an easy way to detect this that I did not find yet

**Reply by moklick** [reply]
Inside a node you can check if a handle is already connected:

```js
const isConnected = useStore(useCallback(s => s.edges.find(n => n.target === id || n.source === nodeId)), [nodeId]);
```

**Reply by devbro1** [reply]
What I meant was how can we detect an edge is in the process of being connected:
![image](https://user-images.githubusercontent.com/29712935/180889027-1207675e-b7fb-460c-9299-c1f231977e68.png)

where edge.startNode = Node and edge.endNode = null.

**Reply by moklick** [reply]
we do have a `onConnectStart` handler. Is this what you are looking for?

**Reply by devbro1** [reply]
This literally resolved my problem. thank you for your direction.

1. I used useContext to pass value to all nodes:
```
const onConnectStart = useCallback((events, params) => {
        context.update({ key: AppContextActionKeyEnum.reactFlowIsDraggingFrom, value: params.handleType });
    }, []);
```
2. I told my nodes to have bigger handles only during a connection being made:
```
function handles() {
        let classes_source = 'w-2 h-2';
        let classes_target = 'w-2 h-2';
        if (context.reactFlowIsDraggingFrom == 'source') {
            classes_target = 'w-full h-full rounded-none opacity-25';
        } else if (context.reactFlowIsDraggingFrom == 'target') {
            classes_source = 'w-full h-full rounded-none opacity-25';
        }
        return (
            <>
                {props.type !== WorkflowNodeEnum.input ? (
                    <Handle type="target" className={classes_target} position={Position.Top} />
                ) : null}
                {props.type !== WorkflowNodeEnum.output ? (
                    <Handle type="source" className={classes_source} position={Position.Bottom} />
                ) : null}
            </>
        );
    }
```
